packages/smooth_app/lib/pages/product/helpers/compatibility_score.dart

```diff
@@ -0,0 +1,644 @@
+import 'dart:io';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/preferences/user_preferences.dart';
+import 'package:smooth_app/data_models/up_to_date_mixin.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/generic_lib/bottom_sheets/smooth_bottom_sheet.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/generic_lib/svg_icon_chip.dart';
+import 'package:smooth_app/helpers/analytics_helper.dart';
+import 'package:smooth_app/helpers/attributes_card_helper.dart';
+import 'package:smooth_app/helpers/haptic_feedback_helper.dart';
+import 'package:smooth_app/knowledge_panel/knowledge_panels/knowledge_panel_page.dart';
+import 'package:smooth_app/knowledge_panel/knowledge_panels_builder.dart';
+import 'package:smooth_app/pages/navigator/app_navigator.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_page.dart';
+import 'package:smooth_app/pages/product/product_incomplete_card.dart';
+import 'package:smooth_app/pages/product/product_page/new_product_page.dart';
+import 'package:smooth_app/resources/app_icons.dart' as icons;
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_app/themes/smooth_theme_colors.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
+import 'package:smooth_app/widgets/smooth_scaffold.dart';
+
+const List<String> _ATTRIBUTE_IDS = <String>[
+  // Nutritional quality
+  Attribute.ATTRIBUTE_NUTRISCORE,
+  Attribute.ATTRIBUTE_LOW_SALT,
+  Attribute.ATTRIBUTE_LOW_SUGARS,
+  Attribute.ATTRIBUTE_LOW_FAT,
+  Attribute.ATTRIBUTE_LOW_SATURATED_FAT,
+  // Ingredients
+  Attribute.ATTRIBUTE_VEGAN,
+  Attribute.ATTRIBUTE_VEGETARIAN,
+  Attribute.ATTRIBUTE_PALM_OIL_FREE,
+  // Environment
+  Attribute.ATTRIBUTE_ECOSCORE,
+  Attribute.ATTRIBUTE_FOREST_FOOTPRINT,
+  // Food processing
+  Attribute.ATTRIBUTE_NOVA,
+  Attribute.ATTRIBUTE_ADDITIVES,
+  // Labels
+  Attribute.ATTRIBUTE_LABELS_ORGANIC,
+  Attribute.ATTRIBUTE_LABELS_FAIR_TRADE,
+  // Allergens
+  Attribute.ATTRIBUTE_ALLERGENS_NO_GLUTEN,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_MILK,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_EGGS,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_NUTS,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_PEANUTS,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_SESAME_SEEDS,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_SOYBEANS,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_CELERY,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_MUSTARD,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_LUPIN,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_FISH,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_CRUSTACEANS,
+  Attribute.ATTRIBUTE_ALLERGENS_NO_SULPHUR_DIOXIDE_AND_SULPHITES,
+];
+
+class CompatibilityScore extends StatefulWidget {
```

Comment:
A single Widget for all that stuff is a bad Flutter behavior.
Please split your code into multiple widgets.